books/kestrel/utilities/system/numbered-names.lisp

```diff
@@ -123,17 +126,25 @@
   "{")
 
 (define get-numbered-name-index-start ((wrld plist-worldp))
-  :returns (start "A @(tsee numbered-name-index-start-p).")
-  :verify-guards nil
+  :returns (start numbered-name-index-start-p "A @(tsee numbered-name-index-start-p).")
   :parents (numbered-name-index-start)
   :short "Retrieve the current starting marker
           of the numeric index of numbered names."
   :long
   "<p>
    If the starting marker is not set yet, the default is returned.
    </p>"
-  (let ((pair (assoc-eq 'start (table-alist 'numbered-name-index-start wrld))))
-    (if pair (cdr pair) *default-numbered-name-index-start*)))
+  (let ((pair (assoc-eq 'start (table-alist+ 'numbered-name-index-start wrld))))
+    (if (and pair
+             (numbered-name-index-start-p (cdr pair)))
+        (cdr pair)
+      *default-numbered-name-index-start*)))
+
+(defrule stringp-of-get-numbered-name-index-start
```

Comment:
Using `defrule` is just fine. For even more compactness, you may be interested in `defret`.